### ISSUE TITLE
Pacman-mirrors Interactive mode brocken

### ISSUE THREAD

**Issue by scachemaille** [manjaro/pacman-mirrors#123]
I just tried
`sudo pacman-mirros -i`
and it seems broken.
I have the following error:
```
   0.339 Germany        : https://mirror.philpot.de/manjaro/
   0.134 Germany        : http://mirror.philpot.de/manjaro/
Traceback (most recent call last):
  File "/usr/bin/pacman-mirrors", line 57, in <module>
    pm.run()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 912, in run
    self.build_interactive_mirror_list()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 609, in build_interactive_mirror_list
    self.default)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/dialogs/graphicalui.py", line 160, in run
    window = GraphicalUI(server_list, random, default)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/dialogs/graphicalui.py", line 54, in __init__
    "{}h {}m".format(server["last_sync"][:2],
TypeError: 'int' object is not subscriptable
```
pacman-mirrors 4.7.5-1

**Comment by Skycoder42** [comment]
Can confirm this with 4.7.6-2 as well. It's excatly the same error.

**Comment by petsam** [comment]
Same error 
Version 4.7.6

**Comment by fhdk** [comment]
I cant reproduce it.
When the error occurs what is the content of `custom-mirrors.json`, `status.json` from `/var/lib/pacman-mirrors`?

![pacman-mirrors-interactive](https://user-images.githubusercontent.com/22748698/37581498-2e893730-2b49-11e8-90ac-7dd139b89566.png)

**Comment by Skycoder42** [comment]
Here are my files, right after running `sudo pacmam-mirrors -i`. I also added `pacman-mirrors.conf`.

/etc/pacman-mirrors.conf:
```
##
## /etc/pacman-mirrors.conf
##

## Branch Pacman should use (stable, testing, unstable)
Branch = stable

## Generation method
## 1) rank   - rank mirrors depending on their access time
## 2) random - randomly generate the output mirrorlist
# Method = rank

## Specify to use only mirrors from specific a country.
## Can add multiple countries separated by a comma (ex: Germany,France)
## Empty means all
OnlyCountry = Germany

## Mirrors directory
# MirrorlistsDir = /etc/pacman.d/mirrors

## Output file
# OutputMirrorlist = /etc/pacman.d/mirrorlist

## When set to True prevents the regeneration of the mirrorlist if
## pacman-mirrors is invoked with the --no-update argument.
## Useful if you don't want the mirrorlist regenerated after a
## pacman-mirrors package upgrade.
# NoUpdate = False
```

/var/lib/pacman-mirrors/custom-mirrors.json:
```
[
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "00:44",
        "protocols": [
            "http"
        ],
        "resp_time": "00.00",
        "url": "http://mirror.ragenetwork.de/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "04:14",
        "protocols": [
            "https",
            "ftp"
        ],
        "resp_time": "00.00",
        "url": "https://ftp.halifax.rwth-aachen.de/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "09:04",
        "protocols": [
            "http",
            "ftp"
        ],
        "resp_time": "00.00",
        "url": "http://ftp.tu-chemnitz.de/pub/linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:04",
        "protocols": [
            "https"
        ],
        "resp_time": "00.00",
        "url": "https://mirror.netzspielplatz.de/manjaro/packages/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "09:24",
        "protocols": [
            "https"
        ],
        "resp_time": "00.00",
        "url": "https://mirror.netcologne.de/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "04:44",
        "protocols": [
            "https"
        ],
        "resp_time": "00.00",
        "url": "https://repo.rhindon.net/manjaro/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "Germany",
        "last_sync": "9999:99",
        "protocols": [
            "https"
        ],
        "resp_time": "99.99",
        "url": "https://repo.stdout.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:05",
        "protocols": [
            "https"
        ],
        "resp_time": "00.00",
        "url": "https://mirror.alpix.eu/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Germany",
        "last_sync": "10:45",
        "protocols": [
            "http",
            "ftp"
        ],
        "resp_time": "00.00",
        "url": "http://ftp.rz.tu-bs.de/pub/mirror/manjaro.org/repos/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:05",
        "protocols": [
            "https",
            "http"
        ],
        "resp_time": "00.00",
        "url": "https://mirror.philpot.de/manjaro/"
    }
]
```

/var/lib/pacman-mirrors/status.json:
```
[
    {
        "branches": [
            1,
            1,
            0
        ],
        "country": "Australia",
        "last_sync": "21:11",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.ventraip.net.au/Manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Australia",
        "last_sync": "01:11",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.melbourneitmirror.net/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Australia",
        "last_sync": "3338:40",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.mirror.serversaustralia.com.au/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Austria",
        "last_sync": "06:11",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.inode.at/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Bangladesh",
        "last_sync": "05:11",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.xeonbd.com/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Belarus",
        "last_sync": "06:11",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.datacenter.by/pub/mirrors/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Belgium",
        "last_sync": "148:14",
        "protocols": [
            "http",
            "ftp"
        ],
        "url": "http://ftp.belnet.be/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Belgium",
        "last_sync": "06:11",
        "protocols": [
            "https"
        ],
        "url": "https://manjaro.cu.be/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Bulgaria",
        "last_sync": "00:41",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.netix.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Bulgaria",
        "last_sync": "00:14",
        "protocols": [
            "https"
        ],
        "url": "https://manjaro.ipacct.com/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Bulgaria",
        "last_sync": "02:41",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.telecoms.bg/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Canada",
        "last_sync": "00:54",
        "protocols": [
            "https"
        ],
        "url": "https://osmirror.org/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Chile",
        "last_sync": "00:11",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.dcc.uchile.cl/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "China",
        "last_sync": "05:14",
        "protocols": [
            "https",
            "http",
            "ftp"
        ],
        "url": "https://mirrors.shu.edu.cn/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "China",
        "last_sync": "05:11",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.ustc.edu.cn/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "China",
        "last_sync": "05:11",
        "protocols": [
            "http"
        ],
        "url": "http://mirrors.tuna.tsinghua.edu.cn/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "China",
        "last_sync": "05:11",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.zju.edu.cn/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "China",
        "last_sync": "05:15",
        "protocols": [
            "ftp",
            "https"
        ],
        "url": "https://mirrors.shuosc.org/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Colombia",
        "last_sync": "5307:33",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.edatel.net.co/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Costa_Rica",
        "last_sync": "03:42",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.ucr.ac.cr/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Czech",
        "last_sync": "06:12",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.dkm.cz/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Denmark",
        "last_sync": "01:02",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.dotsrc.org/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Denmark",
        "last_sync": "00:02",
        "protocols": [
            "https"
        ],
        "url": "https://www.uex.dk/public/manjaro/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "Ecuador",
        "last_sync": -1,
        "protocols": [
            "https"
        ],
        "url": "https://mirror.cedia.org.ec/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Ecuador",
        "last_sync": "01:33",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.espoch.edu.ec/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "France",
        "last_sync": "29:43",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.lignux.com/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "France",
        "last_sync": "05:13",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.free.org/mirrors/repo.manjaro.org/repos/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "France",
        "last_sync": "00:53",
        "protocols": [
            "https"
        ],
        "url": "https://fr.mirror.babylon.network/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "France",
        "last_sync": "00:13",
        "protocols": [
            "http"
        ],
        "url": "http://kibo.remi.lu/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:43",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.ragenetwork.de/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:13",
        "protocols": [
            "ftp",
            "https"
        ],
        "url": "https://ftp.halifax.rwth-aachen.de/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "05:03",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.tu-chemnitz.de/pub/linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:03",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.netzspielplatz.de/manjaro/packages/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "05:13",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.netcologne.de/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "06:43",
        "protocols": [
            "https"
        ],
        "url": "https://repo.rhindon.net/manjaro/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "Germany",
        "last_sync": -1,
        "protocols": [
            "https"
        ],
        "url": "https://repo.stdout.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:06",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.alpix.eu/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "02:16",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.rz.tu-bs.de/pub/mirror/manjaro.org/repos/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Germany",
        "last_sync": "00:03",
        "protocols": [
            "http",
            "https"
        ],
        "url": "https://mirror.philpot.de/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Greece",
        "last_sync": "00:26",
        "protocols": [
            "ftp",
            "http",
            "https"
        ],
        "url": "https://ftp.cc.uoc.gr/mirrors/linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Hong_Kong",
        "last_sync": "09:13",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.cuhk.edu.hk/pub/Linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Hungary",
        "last_sync": "06:13",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.infotronik.hu/mirrors/pub/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            0
        ],
        "country": "Indonesia",
        "last_sync": "20:14",
        "protocols": [
            "http"
        ],
        "url": "http://kambing.ui.ac.id/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Indonesia",
        "last_sync": "03:04",
        "protocols": [
            "http"
        ],
        "url": "http://kartolo.sby.datautama.net.id/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Iran",
        "last_sync": "04:47",
        "protocols": [
            "https",
            "http"
        ],
        "url": "https://repo.sadjad.ac.ir/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Ireland",
        "last_sync": "05:44",
        "protocols": [
            "ftp",
            "https"
        ],
        "url": "https://ftp.heanet.ie/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Italy",
        "last_sync": "10:04",
        "protocols": [
            "https"
        ],
        "url": "https://manjaro.mirror.garr.it/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Italy",
        "last_sync": "10:04",
        "protocols": [
            "https"
        ],
        "url": "https://ba.mirror.garr.it/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Italy",
        "last_sync": "10:04",
        "protocols": [
            "https"
        ],
        "url": "https://ct.mirror.garr.it/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Japan",
        "last_sync": "01:14",
        "protocols": [
            "http"
        ],
        "url": "http://ftp.tsukuba.wide.ad.jp/Linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            0
        ],
        "country": "Japan",
        "last_sync": "12:14",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.riken.jp/Linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "02:54",
        "protocols": [
            "ftp",
            "http",
            "https"
        ],
        "url": "https://mirror.neostrada.nl/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "04:44",
        "protocols": [
            "ftp",
            "https"
        ],
        "url": "https://ftp.nluug.nl/pub/os/Linux/distr/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "06:14",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.snt.utwente.nl/pub/linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "00:54",
        "protocols": [
            "https"
        ],
        "url": "https://nl.mirror.babylon.network/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "04:54",
        "protocols": [
            "https"
        ],
        "url": "https://manjaro.mirror.wearetriple.com/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Netherlands",
        "last_sync": "04:44",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.koddos.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            0
        ],
        "country": "Philippines",
        "last_sync": "13:14",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.rise.ph/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Poland",
        "last_sync": "00:54",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.chmuri.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Poland",
        "last_sync": "00:14",
        "protocols": [
            "https",
            "http"
        ],
        "url": "https://mirror.tuchola-dc.pl/manjaro/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "Portugal",
        "last_sync": "3415:14",
        "protocols": [
            "http",
            "ftp"
        ],
        "url": "http://ftp.dei.uc.pt/pub/linux/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Portugal",
        "last_sync": "00:44",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.barata.pt/"
    },
    {
        "branches": [
            1,
            1,
            0
        ],
        "country": "Romania",
        "last_sync": "12:14",
        "protocols": [
            "http"
        ],
        "url": "http://mirrors.serverhost.ro/manjaro/packages/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Russia",
        "last_sync": "00:04",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.yandex.ru/mirrors/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Russia",
        "last_sync": "03:04",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.truenetwork.ru/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Singapore",
        "last_sync": "05:14",
        "protocols": [
            "https"
        ],
        "url": "https://download.nus.edu.sg/mirror/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "South_Africa",
        "last_sync": "03:15",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.is.co.za/mirrors/manjaro.org/"
    },
    {
        "branches": [
            0,
            0,
            0
        ],
        "country": "South_Africa",
        "last_sync": "21:45",
        "protocols": [
            "http",
            "ftp"
        ],
        "url": "http://manjaro.mirror.ac.za/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Sweden",
        "last_sync": "00:45",
        "protocols": [
            "ftp",
            "https"
        ],
        "url": "https://ftp.lysator.liu.se/pub/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Spain",
        "last_sync": "04:05",
        "protocols": [
            "http"
        ],
        "url": "http://ftp.caliu.cat/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Sweden",
        "last_sync": "02:38",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.zetup.net/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Taiwan",
        "last_sync": "04:05",
        "protocols": [
            "http"
        ],
        "url": "http://free.nchc.org.tw/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "Turkey",
        "last_sync": "06:15",
        "protocols": [
            "ftp",
            "http"
        ],
        "url": "http://ftp.linux.org.tr/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_Kingdom",
        "last_sync": "01:15",
        "protocols": [
            "http"
        ],
        "url": "http://mirror.catn.com/pub/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_Kingdom",
        "last_sync": "00:35",
        "protocols": [
            "http"
        ],
        "url": "http://repo.manjaro.org.uk/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_Kingdom",
        "last_sync": "04:48",
        "protocols": [
            "https"
        ],
        "url": "https://www.mirrorservice.org/sites/repo.manjaro.org/repos/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_Kingdom",
        "last_sync": "00:48",
        "protocols": [
            "http"
        ],
        "url": "http://manjaro.mirrors.uk2.net/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "United_States",
        "last_sync": -1,
        "protocols": [
            "http"
        ],
        "url": "http://mirror.dacentec.com/manjaro/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "United_States",
        "last_sync": -1,
        "protocols": [
            "http"
        ],
        "url": "http://mirror.solarvps.com/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_States",
        "last_sync": "01:08",
        "protocols": [
            "http"
        ],
        "url": "http://distro.ibiblio.org/manjaro/"
    },
    {
        "branches": [
            -1,
            -1,
            -1
        ],
        "country": "United_States",
        "last_sync": -1,
        "protocols": [
            "https"
        ],
        "url": "https://mirror.clarkson.edu/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_States",
        "last_sync": "00:49",
        "protocols": [
            "https"
        ],
        "url": "https://mirrors.ocf.berkeley.edu/manjaro/"
    },
    {
        "branches": [
            1,
            1,
            1
        ],
        "country": "United_States",
        "last_sync": "00:35",
        "protocols": [
            "https"
        ],
        "url": "https://mirror.math.princeton.edu/pub/manjaro/"
    }
]
```

**Comment by Skycoder42** [comment]
Here are my files, right after running `sudo pacmam-mirrors -i`. I also added `pacman-mirrors.conf`.

- [pacman.conf](https://github.com/manjaro/pacman-mirrors/files/1824532/pacman.conf.txt)
- [custom-mirrors.json.txt](https://github.com/manjaro/pacman-mirrors/files/1824534/custom-mirrors.json.txt)
- [status.json.txt](https://github.com/manjaro/pacman-mirrors/files/1824535/status.json.txt)

**Comment by fhdk** [comment]
Your `pacman-mirrors.conf` contains invalid settings. The current valid configuration is [here](https://github.com/manjaro/pacman-mirrors/blob/master/conf/pacman-mirrors.conf)

The other files is OK.

Using the command mentioned in OP

    sudo pacman-mirrors -i

I cannot reproduce the issue. Not even with a full mirror pool. (`pacman-mirrors -c all`)

**Comment by petsam** [comment]
My `pacman-mirrors.conf`
```
##
#/etc/pacman-mirrors.conf
##

#Branch Pacman should use (stable, testing, unstable)
Branch = testing

#Generation method
#1) rank   - rank mirrors depending on their access time
#2) random - randomly generate the output mirrorlist
Method = rank

#Define protocols and priority
#  separated by comma 'https,http' or 'http,https'
#ATM available protocols are: http, https, ftp
#Not specifying a protocol will ban the protocol from being used
#If a mirror has more than one protocol defined only the first is written to the mirrorlist
#Empty means all in reversed alphabetic order
Protocols =

#Get a list of all available counties with 'pacman-mirrors -l'
#Value can be 'Custom' or nothing which means all
# OnlyCountry = 

#When set to False - all certificates are accepted.
#Use only if you fully trust all ssl-enabled mirrors.
SSLVerify = True
```

**Comment by fhdk** [comment]
@Skycoder42 
Are you running on Wayland?

Is it on TTY?

**Comment by fhdk** [comment]
I have made a `pacman-mirrors-dev` package. It is in unstable.

Install it with 

    sudo pacman -Syu https://mirror.netzspielplatz.de/manjaro/packages/unstable/core/x86_64/pacman-mirrors-dev-4.8.1-1-any.pkg.tar.xz

Let me know if it solves the issue.

**Comment by xircon** [comment]
Same error.  Plasma xorg.
[first.zip](https://github.com/manjaro/pacman-mirrors/files/1825984/first.zip)
[second.zip](https://github.com/manjaro/pacman-mirrors/files/1825985/second.zip)

First - before interactive
Second - After.

**Comment by xircon** [comment]
Just updated to pacman-mirrors-4.8.1-3 - still crashing in interactive mode.

**Comment by fhdk** [comment]
@xircon 
Could you post the error message from 4.8.1-3?

**Comment by commonsourcecs** [comment]
Failing for me with this return:
```
Traceback (most recent call last):
  File "/usr/bin/pacman-mirrors", line 57, in <module>
    pm.run()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 912, in run
    self.build_interactive_mirror_list()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 609, in build_interactive_mirror_list
    self.default)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/dialogs/graphicalui.py", line 160, in run
    window = GraphicalUI(server_list, random, default)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/dialogs/graphicalui.py", line 54, in __init__
    "{}h {}m".format(server["last_sync"][:2],
TypeError: 'int' object is not subscriptable
```
[pacman-mirrors.tar.gz](https://github.com/manjaro/pacman-mirrors/files/1826040/pacman-mirrors.tar.gz)
I had run '-i' before creating the first archive already, but here is one after command fail again:
[pacman-mirrors2.tar.gz](https://github.com/manjaro/pacman-mirrors/files/1826056/pacman-mirrors2.tar.gz)

**Comment by fhdk** [comment]
@xircon 
@commonsourcecs

That is most weird - that code does not exist `"{}h {}m".format(server["last_sync"][:2],`

Try installing this

[pacman-mirrors-4.8.1-4.tar.gz](https://github.com/manjaro/pacman-mirrors/files/1826100/pacman-mirrors-4.8.1-4.tar.gz)

**Comment by fhdk** [comment]
The lines that errors are these

https://github.com/manjaro/pacman-mirrors/blob/0ab1f1c438eeed43fa96616d3e07c2deff9e75bc/pacman_mirrors/dialogs/graphicalui.py#L49-L58

**Comment by commonsourcecs** [comment]
Found source for me. 
'-i' fails when there is any discrepancy in the list. In my region/list, there is one mirror currently Timed Out. 
To test, I switched to Netherlands (all mirrors active and synced to Testing). Then ran '-i' on its own again.
That worked.

**Comment by xircon** [comment]
Had to nip out - 4.8.1-5 same problems - do you want the zip?

**Comment by xircon** [comment]
Yes, mirror times out:
```
   0.213 Germany        : https://mirror.netcologne.de/manjaro/
   ..... Germany        : https://repo.rhindon.net/manjaro/
   ..... Germany        : https://repo.stdout.net/manjaro/
.: ERROR [Errno -2] Name or service not known 'https://repo.stdout.net/manjaro/'
```
Could be the problem?

**Comment by bogdancovaciu** [comment]
Hello,
Finally i'm here too, even if i feel that is way too overwhelming. Here are the files you requested on manjaro forum:
[pacman_mirrors_1.zip](https://github.com/manjaro/pacman-mirrors/files/1826288/pacman_mirrors_1.zip)
[pacman_mirrors_2.zip](https://github.com/manjaro/pacman-mirrors/files/1826289/pacman_mirrors_2.zip)

Are in the order you requested and hopefully helpful!

**Comment by commonsourcecs** [comment]
4.8.1-1 also
Confirmed that '-i' fails with list containing 'Timed Out' mirrors but succeeds on list without. 
('--api' or '-B' no longer works though)

**Comment by fhdk** [comment]
Yes - the api is my bad - I added some checks for any api related argument eg -B without the actual `--api` present - and it failed me. This has been corrected.

https://www.uex.dk/public/manjaro/unstable/core/x86_64/pacman-mirrors-4.8.1-6-any.pkg.tar.xz

**Comment by Skycoder42** [comment]
With version `Version 4.8.1dev` (4.8.1-6) the error message has changed a little:

```
Traceback (most recent call last):
  File "/usr/bin/pacman-mirrors", line 57, in <module>
    pm.run()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 853, in run
    self.build_interactive_mirror_list()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 585, in build_interactive_mirror_list
    interactive_list = mirrorFn.translate_pool_to_interactive(worklist)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/mirrors/mirrorFn.py", line 207, in translate_pool_to_interactive
    "last_sync": "{}h {}m".format(mirror["last_sync"][:2].replace(":", ""),
TypeError: 'int' object is not subscriptable
```

I too have the mirror that times out - I will try to remove it to test if that solves the problem

**Comment by fhdk** [comment]
The issue has been solved with the release of 4.8.2

**Comment by commonsourcecs** [comment]
4.8.2-2 
```
Traceback (most recent call last):
  File "/usr/bin/pacman-mirrors", line 57, in <module>
    pm.run()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 853, in run
    self.build_interactive_mirror_list()
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/pacman_mirrors.py", line 585, in build_interactive_mirror_list
    interactive_list = mirrorFn.translate_pool_to_interactive(worklist)
  File "/usr/lib/python3.6/site-packages/pacman_mirrors/mirrors/mirrorFn.py", line 213, in translate_pool_to_interactive
    "last_sync": "{}h {}m".format(mirror["last_sync"][:2].replace(":", ""),
TypeError: 'int' object is not subscriptable
```
